### PR TITLE
RF_01.002-01 | FreestyleProject > Rename Project > User receives an Error when the new name is invalid

### PR DESCRIPTION
**Pull request by obannikova**
RF: https://github.com/RedRoverSchool/JenkinsQA_JS_2024_fall/issues/588

TC: https://github.com/RedRoverSchool/JenkinsQA_JS_2024_fall/issues/93

US: https://github.com/RedRoverSchool/JenkinsQA_JS_2024_fall/issues/21

### CODE REVIEW

File: cypress/e2e/freestyleProjectRenameProjectPOM.cy.js [hunk 1]
```diff
@@ -35,88 +35,173 @@ describe("US_01.002 | FreestyleProject > Rename Project", () => {
...
+      .clickRenameButtonSubmit();
+
+    freestyleProjectPage.getJobHeadline().should("have.text", project.newName);
   });
-    
-  it('TC_01.002.11 | Rename a project name from the Project Page', () => {
+
+  it("TC_01.002-01 | User receives an Error when the new name is invalid", () => {
+    let project1Name = faker.commerce.productName();
+    let project2Name = faker.commerce.productName();
```
Comment (Negrescu29) [inline cypress/e2e/freestyleProjectRenameProjectPOM.cy.js:112]: Instead of importing faker and creating these 2 variables you can use the genData.js from `redRover/JenkinsQA_JS_2024_fall/cypress/fixtures/genData.js` which will give you the same project name

`let Project = {
            name: faker.company.buzzNoun(),
            newName: faker.company.buzzNoun(),`

File: cypress/e2e/freestyleProjectRenameProjectPOM.cy.js [hunk 2]
```diff
@@ -35,88 +35,173 @@ describe("US_01.002 | FreestyleProject > Rename Project", () => {
...
+      .clickOKButton();
+    header.clickJenkinsLogo();
+
+    cy.log("precondition 2: creating project2");
+    dashboardPage.clickNewItemMenuLink();
+    newJobPage
+      .typeNewItemName(project2Name)
+      .selectFreestyleProject()
+      .clickOKButton();
+    header.clickJenkinsLogo();
```
Comment (Negrescu29) [inline cypress/e2e/freestyleProjectRenameProjectPOM.cy.js:128]: @obannikova could you explain the logic behind creating 2 different projects?
Can't you do the checks on one project?

**Review by antonbloha — APPROVED** [review]

File: cypress/e2e/freestyleProjectRenameProjectPOM.cy.js (hunk 1 above)
Comment (obannikova) [reply cypress/e2e/freestyleProjectRenameProjectPOM.cy.js]: I've replaced the direct use of faker with genData.js

File: cypress/e2e/freestyleProjectRenameProjectPOM.cy.js (hunk 2 above)
Comment (obannikova) [reply cypress/e2e/freestyleProjectRenameProjectPOM.cy.js]: I need two projects to check the scenario: "step3: rename the project with the name of the existing project". For that I need two projects

Comment (Negrescu29) [reply cypress/e2e/freestyleProjectRenameProjectPOM.cy.js]: I think you can use one project here, and do something like clear the folder name and then use the same project name, if we have a restriction on the app to not being able to rename using existing folder name it doesn't really matter what folder name you use.

Comment (Negrescu29) [reply cypress/e2e/freestyleProjectRenameProjectPOM.cy.js]: you have 100 lines of code for your test and personally, it's hard to follow for me, this is all that I'm saying.

Comment (obannikova) [reply cypress/e2e/freestyleProjectRenameProjectPOM.cy.js]: Got it, I will make the next scenarios shorter)

**Review by irinadash — APPROVED** [review]
